core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -852,6 +854,11 @@ public Index(StreamInput in) throws IOException {
             this.ttl = in.readLong();
             this.versionType = VersionType.fromValue(in.readByte());
             assert versionType.validateVersionForWrites(this.version);
+            if (format >= FORMAT_AUTOGENERATED_IDS) {
```

Comment:
Not part of your change, but I do wonder how that `assert format == SERIALIZATION_FORMAT : "format was: " + format;` assertion actually work with our BWC tests...